### PR TITLE
add install script

### PR DESCRIPTION
**Pull request by jairad26**

### CODE REVIEW

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:121]:
```suggestion
    echo -e "set -gx HYP_CLI \"$install_dir\"\nstring match -r \".hypermode\" \"\$PATH\" > /dev/null; or set -gx PATH \"\$HYP_CLI/bin\" \$PATH"
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:124]:
```suggestion
    echo -e "\n# ${CLI_NAME}\nexport HYP_CLI=\"$install_dir\"\nexport PATH=\"\$HYP_CLI/bin:\$PATH\""
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:140]:
```suggestion
    printf "[3/4] ${DIM}The ${CLI_NAME} is already in \$PATH${RESET}\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:12]:
```suggestion
# Config
CLI_NAME="Hypermode CLI"
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:154]:
```suggestion
    printf "[3/4] ${DIM}The ${CLI_NAME} has already been configured in ${PROFILE}${RESET}\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:161]:
```suggestion
  printf "[3/4] ${DIM}Added the ${CLI_NAME} to the PATH in ${PROFILE}${RESET}\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:178]:
```suggestion
    update_profile "$INSTALL_DIR" && printf "[4/4] ${DIM}Installed the ${CLI_NAME}${RESET}\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:183]:
```suggestion
  printf "[1/4] ${DIM}Downloading the ${CLI_NAME} from NPM${RESET}\n"
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:193]:
```suggestion
    printf "Could not download the ${CLI_NAME} version '$VERSION' from\n$url\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:198]:
```suggestion
  printf "[1/4] ${DIM}Downloaded latest ${CLI_NAME} v${VERSION}${RESET}\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:274]:
```suggestion
printf "\n${BOLD}${BLUE}${CLI_NAME}${RESET} Installer ${DIM}v${INSTALLER_VERSION}${RESET}\n\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [inline install.sh:280]:
```suggestion
printf "\nThe ${CLI_NAME} has been installed! 🎉\n" >&2
```

File: install.sh
Comment (mattjohnsonpint) [reply install.sh:12]: @ryanfoxtyler should decide it it's "Hypermode CLI" or "Hyp CLI" before we merge this, but I like having it in a single variable.

**Review by mattjohnsonpint — CHANGES_REQUESTED** [review]

File: install.sh
Comment (mattjohnsonpint) [reply install.sh:12]: I'll update the Modus installer with that too.

Comment (ryanfoxtyler) [reply install.sh:12]: Hyp CLI

Comment (mattjohnsonpint) [reply install.sh:12]: Ryan said "Hyp CLI". 👍

**Review by mattjohnsonpint — APPROVED** [review]

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**
